spec/knackhq/client_spec.rb

```diff
@@ -0,0 +1,65 @@
+require 'spec_helper'
+
+describe Knackhq::Client do
+  it 'has a version number' do
+    expect(Knackhq::Client::VERSION).not_to be nil
+  end
+
+  let(:client) { knack_client }
+  let(:base_uri) { 'http://example.com' }
+  let(:x_knack_application_id) { '00000000' }
+  let(:x_knack_rest_api_key) { '99999999' }
+
+  let(:knack_client) do
+    Knackhq::Client.new(base_uri,
+                        x_knack_application_id,
+                        x_knack_rest_api_key)
+  end
+
+  describe '#new' do
+    it { expect(knack_client).to be_a Knackhq::Client }
+  end
+
+  describe '#request.get' do
```

Comment:
I think I get what you are saying. that way, an object call can have a post or a get associated with it.
If I have you correctly: 
```Ruby
module Knackhq
  class Client
    def request
      headers = { 'x-knack-application-id' => @x_knack_application_id.dup,
                  'Content-Type' => 'application/json',
                  'x-knack-rest-api-key' => @x_knack_rest_api_key.dup }
      def all_objects
        hash_request = objects.get.to_h[:objects]
        hash_request.map do |hash|
          Hashie.symbolize_keys! hash
        end
      end
def object(:key)
  # request is private
  # I don't know if this is a valid blanket request, but its for demonstration purposes.
  request.get.objects(:key)
end
private
      Blanket.wrap(@base_uri.dup,
                   :headers => headers)
    end

    def objects
       request
    end